vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/DefaultHyperlinkCellClickHandlerTests.java

```diff
@@ -0,0 +1,57 @@
+package com.vaadin.flow.component.spreadsheet.tests;
+
+import static org.junit.Assert.*;
+
+import java.io.File;
+import java.net.URL;
+
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+import org.junit.Test;
+
+import com.vaadin.flow.component.spreadsheet.DefaultHyperlinkCellClickHandler;
+import com.vaadin.flow.component.spreadsheet.Spreadsheet;
+
+/**
+ * Created by mtzukanov on 24.4.2017.
+ */
+public class DefaultHyperlinkCellClickHandlerTests {
+
+    @Test
+    public void hyperlinkParser_validStrings_correctParsed() throws Exception {
+        URL testSheetResource = this.getClass().getClassLoader()
+                .getResource("test_sheets/hyper_links.xlsx");
+        File testSheetFile = new File(testSheetResource.toURI());
+
+        Workbook workbook = WorkbookFactory.create(testSheetFile);
+
+        final Spreadsheet ss = new Spreadsheet(workbook);
+        ss.setActiveSheetIndex(0);
+
+        TestHyperlinkCellClickHandler handler = new TestHyperlinkCellClickHandler(
+                ss);
+
+        // this tests the condition from #537, formula first argument is a cell
+        // ref whose value is the link target
```

Comment:
Nit. This also links to an unrelated issue. This is the right one (https://github.com/vaadin/spreadsheet/pull/537)